subcommand/connect-init/command_test.go

```diff
@@ -45,140 +45,230 @@ func TestRun_FlagValidation(t *testing.T) {
-	proxyFile := common.WriteTempFile(t, "")
-	tokenFile := common.WriteTempFile(t, "")
-
-	// Start Consul server with ACLs enabled and default deny policy.
-	var masterToken = "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
-	server, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
-		c.ACL.Enabled = true
-		c.ACL.DefaultPolicy = "deny"
-		c.ACL.Tokens.Master = masterToken
-	})
-	defer server.Stop()
-	require.NoError(t, err)
-	server.WaitForLeader(t)
-	consulClient, err := api.NewClient(&api.Config{Address: server.HTTPAddr, Token: masterToken})
-	require.NoError(t, err)
-
-	// Start the mock k8s server.
-	k8sMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("content-type", "application/json")
-		if r != nil && r.URL.Path == "/apis/authentication.k8s.io/v1/tokenreviews" && r.Method == "POST" {
-			w.Write([]byte(tokenReviewFoundResponse))
-		}
-		if r != nil && r.URL.Path == "/api/v1/namespaces/default/serviceaccounts/counting" && r.Method == "GET" {
-			w.Write([]byte(readServiceAccountFound))
-		}
-	}))
-	defer k8sMockServer.Close()
-
-	// Set up Consul's auth method.
-	authMethodTmpl := api.ACLAuthMethod{
-		Name:        testAuthMethod,
-		Description: "Kubernetes Auth Method",
-		Type:        "kubernetes",
-		Config: map[string]interface{}{
-			"Host":              k8sMockServer.URL,
-			"CACert":            serviceAccountCACert,
-			"ServiceAccountJWT": serviceAccountJWTToken,
+	cases := []struct {
+		name   string
+		secure bool
+	}{
+		{
+			name:   "ACLs enabled, not secure",
+			secure: false,
+		},
+		{
+			name:   "ACLs enabled, secure",
+			secure: true,
 		},
 	}
-	_, _, err = consulClient.ACL().AuthMethodCreate(&authMethodTmpl, nil)
-	require.NoError(t, err)
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			bearerFile := common.WriteTempFile(t, serviceAccountJWTToken)
+			proxyFile := common.WriteTempFile(t, "")
+			tokenFile := common.WriteTempFile(t, "")
 
-	// Create the binding rule.
-	aclBindingRule := api.ACLBindingRule{
-		Description: "Kubernetes binding rule",
-		AuthMethod:  testAuthMethod,
-		BindType:    api.BindingRuleBindTypeService,
-		BindName:    "${serviceaccount.name}",
-		Selector:    "serviceaccount.name!=default",
-	}
-	_, _, err = consulClient.ACL().BindingRuleCreate(&aclBindingRule, nil)
-	require.NoError(t, err)
+			var caFile, certFile, keyFile string
+			// Start Consul server with ACLs enabled and default deny policy.
+			var masterToken = "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
+			server, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+				c.ACL.Enabled = true
+				c.ACL.DefaultPolicy = "deny"
+				c.ACL.Tokens.Master = masterToken
+				if test.secure {
+					caFile, certFile, keyFile = common.GenerateServerCerts(t)
+					c.CAFile = caFile
+					c.CertFile = certFile
+					c.KeyFile = keyFile
+				}
+			})
+			defer server.Stop()
+			require.NoError(t, err)
```

Comment:
```suggestion
			require.NoError(t, err)
			defer server.Stop()
```

`server` could be nil if `err != nil` which would result in panic

Reply:
I saw that below as well

Reply:
Great catch, found this 3 other places! thx!